compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -1307,7 +1307,11 @@ fn ice_path() -> &'static Option<PathBuf> {
 /// internal features.
 ///
 /// A custom rustc driver can skip calling this to set up a custom ICE hook.
-pub fn install_ice_hook(
+///
+/// # Unsafety
+///
+/// This function modifies the environment and has the same safety as [`std::env::set_var`].
+pub unsafe fn install_ice_hook(
```

Comment:
If this is only for setting RUST_BACKTRACE, maybe use https://doc.rust-lang.org/nightly/std/panic/fn.set_backtrace_style.html instead.

In general I think we should separate "making the compiler not trigger the new lint" from the PR that introduces the lint, or else review will easily get terribly derailed. (For instance, with such a large unsafe fn, we'd definitely want to enable `unsafe_op_in_unsafe_fn` as well.)

Reply:
> If this is only for setting RUST_BACKTRACE, maybe use https://doc.rust-lang.org/nightly/std/panic/fn.set_backtrace_style.html instead.

Done.

> In general I think we should separate "making the compiler not trigger the new lint" from the PR that introduces the lint, or else review will easily get terribly derailed. (For instance, with such a large unsafe fn, we'd definitely want to enable `unsafe_op_in_unsafe_fn` as well.)

Factored out into #125063.